### DISCUSSION TITLE
Set source and global package path for project via custom target in a nuget package

### DISCUSSION THREAD

**Discussion by nisha-ashwin** [NuGet/Home#11225] [Q&A]
Hi,

I am creating a nuget package which will set the nuget source and global packages path for project which consumes it.
I want to set this property before the restore target so that the packages are restored from the said sources and downloaded at the said packages folder.
I have tried to call a custom target before the _GetRestoreSettings target to set the properties. But it doesn't get called by the project that is consuming my custom target nuget package.

Could anyone tell me when should I call my custom target to set the above properties? I am building the project using dotnet cli. 

Thanks in advance.

**Comment by erdembayar** [comment]
Thank you for asking this question.
Actually package is mutating (setting GPF and source) restore context is not desirable. If another package start doing same thing(setting own GPF and source) then which one get respected?
Imagine a project uses this customer's package, and also a package only on the feed the package adds. Now, run dotnet nuget locals global-packages --clean, and then dotnet restore. The package that adds the second source isn't yet installed. How can restore possibly succeed when the second package isn't on the sources known in nuget.config?
